sympy/core/relational.py

```diff
@@ -426,6 +426,43 @@ def __new__(cls, lhs, rhs=None, **options):
                     isinstance(rhs, Boolean)):
                 return S.false  # only Booleans can equal Booleans
 
+            # check re/im parts *quickly* but not as naively as
+            # with pure_complex
+            if lhs.is_number and rhs.is_number:
+                def ri(e):
+                    # treat e like Add and return real and imag parts
+                    # or None (if any term is unknown)
+                    r = []
+                    i = []
+                    for x in Add.make_args(e):
+                        if x is S.ComplexInfinity:
+                            i.append(x)
+                            continue
+                        c = x.coeff(S.ImaginaryUnit)
+                        if c:
+                            if not c.is_extended_real:
+                                return
```

Comment:
This line is also apparently not covered. How would we get here?

Something like this maybe:
```julia
In [7]: x = I*exp(I+oo)                                                                                                           

In [8]: Eq(x, I*x)                                                                                                                
Out[8]: 
     ⅈ       ⅈ
∞⋅ⅈ⋅ℯ  = -∞⋅ℯ 
```
(I don't think it's particularly important to be able to evaluate things like that. Not having a wrong result matters but I can't imagine evaluating the above being particularly useful)

Perhaps a more realistic example:
```julia
In [1]: x, y, z = symbols('x, y, z', infinite=True)                                                                               

In [2]: Eq(x+I*y, x+I*z)                                                                                                          
Out[2]: False
```

Reply:
That last result should be `Eq(y, z)`, shouldn't it? This PR doesn't fix that.

All lines that I added are now covered with the previous or added tests.

Reply:
Normally when an Eq doesn't evaluate it doesn't simplify in any way:
```julia
In [6]: Eq(x+y, x+z)                                                                                                              
Out[6]: x + y = x + z
```

Reply:
And `x - x` is not 0 when you are dealing with infinite quantities.